File: YoutubeRadioApplication/src/main/java/com/ldz/project/service/userregister/inter/IUserRegisterService.java
package com.ldz.project.service.userregister.inter;


import com.ldz.project.model.UserRegister;

import java.util.List;

/**
 * Created by ldalzotto on 15/04/2017.
 */
public interface IUserRegisterService {

    public boolean registerUserFromUserDetails(UserRegister userRegister);

    public List<UserRegister> getDetailsFromusername(String username);

    public UserRegister getCurrentUserFromIpaddress(String ipaddress);

    public UserRegister addUserFromexisting(UserRegister userRegister);

    public void logoutUserFromIpaddress(String ipaddress);

    public UserRegister loginUserFromEmailAndPasswordAndIpaddress(String email, String password,
                                                                     String ipaddress);
}

File: YoutubeRadioApplication/src/scala/com/ldz/project/service/userregister/IUserRegisterService.scala
package com.ldz.project.service.userregister


import com.ldz.project.model.UserRegister

/**
  * Created by ldalzotto on 23/06/2017.
  */
trait IUserRegisterService {
  def registerUserFromUserDetails(userRegister: UserRegister): Boolean

  def getDetailsFromusername(username: String): List[UserRegister]

  def getDetailsFromEmail(email: String): List[UserRegister]

  def getCurrentUserFromIpaddress(ipaddress: String): UserRegister

  def addUserFromexisting(userRegister: UserRegister): UserRegister

  def logoutUserFromIpaddress(ipaddress: String): Unit

  def loginUserFromEmailAndPasswordAndIpaddress(email: String, password: String, ipaddress: String): UserRegister
}
